lib/handler/mruby.c

```diff
@@ -104,11 +104,15 @@ mrb_value h2o_mruby_eval_expr(mrb_state *mrb, const char *expr)
     return mrb_funcall(mrb, mrb_top_self(mrb), "eval", 1, mrb_str_new_cstr(mrb, expr));
 }
 
-void h2o_mruby_define_callback(mrb_state *mrb, const char *name, int id)
+void h2o_mruby_define_callback(mrb_state *mrb, const char *name, h2o_mruby_callback callback)
 {
+    h2o_mruby_shared_context_t *shared_ctx = mrb->ud;
+    h2o_vector_reserve(NULL, &shared_ctx->callbacks, ++shared_ctx->callbacks.size);
+    shared_ctx->callbacks.entries[shared_ctx->callbacks.size - 1] = callback;
```

Comment:
I would appreciate it if you could change the code to the following. It is kind-of aesthetic, but by doing so we can have the guarantee that all the objects stored in the vector are initialized.
```
h2o_vector_reserve(NULL, &shared_ctx->callbacks, shared_ctx->callbacks.size + 1);
shared_ctx->callbacks.entries[shared_ctx->callbacks.size++] = callback;
```